### PR TITLE
Add some more advanced debug features

### PR DESCRIPTION
**Pull request by kbembedded**
Adds some additional debug features.
Refactors logging functionality in to debug, does not write log to file but this can be added back later
Reduces msemu runtime complexity

### CODE REVIEW

File: src/makefile [hunk 1]
```diff
@@ -1,51 +1,26 @@
...
-	$(CC) $(CFLAGS) -c ui.c -o ui.o
-
-debug.o: debug.c
-	$(CC) $(CFLAGS) -c debug.c -o debug.o
-
-clean:
-	rm -f *.o msemu
+
+CC = gcc
+LIBS = -lz80ex -lz80ex_dasm -lSDLmain  -lSDL  -lSDL_gfx
```
Comment (CaptainNic) [inline src/makefile]: Did line endings change?

Comment (kbembedded) [reply src/makefile]: Yeah, makefile was DOS newlines, converted to UNIX in 15e1b30, made actual mods in e7eecb7

Comment (kbembedded) [reply src/makefile]: I guess in e7eecb7 I also committed other documentation changes too. Oops

Comment (kbembedded) [reply src/makefile]: Resolved? Approve to merge?

**Pull request merged**